src/Model/Model.java

```diff
@@ -535,6 +545,68 @@ public boolean sendMessage(Event data) {
         return true;
     }
 
+    public ClaimGrayResult claimGray(ClaimGrayRequest req){
+        ClaimGrayResult res = new ClaimGrayResult();
+        String username = req.getUsername();
+        Game game = getAssociatedGame(username);
+        if (game.claimRoute(username, req.getId())) {
+            Route route = game.getRoute(req.getId());
+            res.setSuccess(true);
+            res.setId(req.getId());
+            if (route.getClaimedType() != null) {
+                res.setColor(route.getClaimedType());
+            }
+            AddEventCommand addEventCommand = new AddEventCommand();
+            Event event = new Event();
+            event.setUsername(username);
+            event.setType(EventType.TURN);
+            event.setContent(req.getUsername() + " claimed the route from " + route.getCity1().getName() + " to " + route.getCity2().getName() + ".");
+            addEventCommand.setEvent(event);
+            CommandData eventCommandData = new CommandData();
+            eventCommandData.setType(ClientCommandType.C_EVENT);
+            eventCommandData.setData(new Gson().toJson(addEventCommand));
+
+            CommandData claimCommandData = new CommandData();
+            claimCommandData.setType(ClientCommandType.C_CLAIM_ROUTE);
+            ClaimRouteCommand claimRouteCommand = new ClaimRouteCommand();
+            claimRouteCommand.setId(req.getId());
+            claimRouteCommand.setUsername(username);
+            claimCommandData.setData(new Gson().toJson(claimRouteCommand));
+
+            CommandData statsCommandData = new CommandData();
+            StatsChange change1 = new StatsChange();
+            StatsChange change2 = new StatsChange();
+            StatsChange change3 = new StatsChange();
+            change1.setType(StatsChangeType.DECREASE_TRAIN_CARS);
+            change1.setAmount(route.getNumTracks());
+            change2.setType(StatsChangeType.ADD_POINTS);
+            change2.setAmount(route.getPoints());
+            change3.setType(StatsChangeType.DECREASE_TRAIN_CAR_CARDS);
+            change3.setAmount(route.getNumTracks());
+            List<StatsChange> changes = new ArrayList<>();
+            changes.add(change1);
+            changes.add(change2);
+            changes.add(change3);
+            UpdatePlayerStatsCommand updatePlayerStatsCommand = new UpdatePlayerStatsCommand();
+            updatePlayerStatsCommand.setUsername(username);
+            updatePlayerStatsCommand.setChanges(changes);
+            statsCommandData.setType(ClientCommandType.C_UPDATE_PLAYER_STATS);
+            statsCommandData.setData(new Gson().toJson(updatePlayerStatsCommand));
+
+            Set<String> usernamesOfPlayers = game.getGamePlayers().keySet();
+            for (String oneUsername: usernamesOfPlayers) {
```

Comment:
Why not use that beautiful function you wrote to add commands to all players in a game? :)